src/connector/src/source/kafka/enumerator/client.rs

```diff
@@ -115,6 +115,69 @@ impl SplitEnumerator for KafkaSplitEnumerator {
 }
 
 impl KafkaSplitEnumerator {
+    pub async fn list_splits_batch(
+        &mut self,
+        expect_start_timestamp_millis: Option<i64>,
+        expect_stop_timestamp_millis: Option<i64>,
+    ) -> anyhow::Result<Vec<KafkaSplit>> {
+        let topic_partitions = self.fetch_topic_partition().await.map_err(|e| {
+            anyhow!(format!(
+                "failed to fetch metadata from kafka ({}), error: {}",
+                self.broker_address, e
+            ))
+        })?;
+        let mut expect_start_offset = if let Some(ts) = expect_start_timestamp_millis {
+            Some(
+                self.fetch_offset_for_time(topic_partitions.as_ref(), ts)
+                    .await?,
+            )
+        } else {
+            None
+        };
+        let mut expect_stop_offset = if let Some(ts) = expect_stop_timestamp_millis {
+            Some(
+                self.fetch_offset_for_time(topic_partitions.as_ref(), ts)
+                    .await?,
+            )
+        } else {
+            None
+        };
+
+        let mut watermarks = {
+            let mut ret = HashMap::new();
+            for partition in &topic_partitions {
+                let (low, high) = self
+                    .client
+                    .fetch_watermarks(self.topic.as_str(), *partition, KAFKA_SYNC_CALL_TIMEOUT)
+                    .await?;
+                ret.insert(partition, (low - 1, high));
+            }
+            ret
+        };
+
+        topic_partitions.iter().map(|partition| {
+            let (low, high) = watermarks.remove(&partition).unwrap();
+            let start_offset = {
+                let start = expect_start_offset.as_mut().map(|m| m.remove(partition).unwrap_or(Some(low))).unwrap_or(Some(low)).unwrap_or(low);
+                i64::max(start, low)
+            };
+            let stop_offset = {
+                let stop = expect_stop_offset.as_mut().map(|m| m.remove(partition).unwrap_or(Some(high))).unwrap_or(Some(high)).unwrap_or(high);
+                i64::min(stop, high)
+            };
+
+            if start_offset > stop_offset {
+                return Err(anyhow!(format!("topic {} partition {}: requested start offset {} is greater than stop offset {}",self.topic, partition, start_offset, stop_offset)));
+            }
+            Ok(KafkaSplit {
+                topic: self.topic.clone(),
+                partition: *partition,
+                start_offset: Some(start_offset),
```

Comment:
If we always return some offset, why not remove the `Option`?

Reply:
To align with other connectors. Still it is a good suggestion, I will take a deeper look